libmamba/src/download/downloader.cpp

```diff
@@ -74,16 +84,62 @@ namespace mamba::download
                         LOG_INFO << "Using REQUESTS_CA_BUNDLE " << remote_fetch_params.ssl_verify;
                     }
                 }
-                else if (remote_fetch_params.ssl_verify == "<system>" && util::on_linux)
+                else if (remote_fetch_params.ssl_verify == "<system>")
                 {
+                    // Use the CA certificates from the `conda-forge::ca-certificates` (a transitive
+                    // dependency of `libmamba` via `libcurl`) which are installed at
+                    // `$CONDA_PREFIX/ssl/cacert.pem` if they exist, otherwise we fallback to the
+                    // system CA certificates.
+
+                    // We cannot assume that the prefix can be known from $CONDA_PREFIX (which
+                    // might not be set), so we infer its value from the path of `libmamba`.
+
+                    fs::u8path libmamba_library_path;
+
+#if defined(__linux__) || __APPLE__ || __MACH__
+                    Dl_info dl_info;
+                    if (dladdr(reinterpret_cast<void*>(init_remote_fetch_params), &dl_info))
+                    {
+                        libmamba_library_path = dl_info.dli_fname;
+                    }
+#else
+                    HMODULE hModule = NULL;
+                    CHAR path[MAX_PATH];
+                    GetModuleHandleEx(
+                        GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS
+                            | GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+                        (LPCTSTR) init_remote_fetch_params,
+                        &hModule
+                    );
+                    GetModuleFileName(hModule, path, MAX_PATH);
+
+                    libmamba_library_path = std::string(path);
+#endif
+                    fs::u8path env_prefix = libmamba_library_path.parent_path().parent_path();
+                    fs::u8path env_prefix_conda_cert = env_prefix / "ssl" / "cacert.pem";
+
+                    LOG_INFO << "Checking for CA certificates at: " << env_prefix_conda_cert;
+
+                    if (fs::exists(env_prefix_conda_cert))
+                    {
+                        LOG_INFO << "Using CA certificates from the environment at: "
+                                 << env_prefix_conda_cert;
+                        remote_fetch_params.ssl_verify = env_prefix_conda_cert;
+                        remote_fetch_params.curl_initialized = true;
+                        return;
+                    }
```

Comment:
I don't think we need all this complexity. As discussed in a meeting, checking if certificates are available under the ROOT_PREFIX sohuld be enough.